web/controllers/users_controller.ex

```diff
@@ -31,14 +31,53 @@ defmodule Thegm.UsersController do
     end
   end
 
-  def show(conn, %{"id" => user_id}) do
+  def update(conn, %{"username" => username, "data" => %{"attributes" => params, "type" => type}}) do
+    current_user_id = conn.assigns[:current_user].id
+    user_id = Users.generate_uuid(username)
+
+    cond do
+      type == "users" ->
+        case Repo.get(Users, user_id) |> Repo.preload([{:group_members, :groups}]) do
+          nil ->
+            conn
+            |> put_status(:not_found)
+            |> render(Thegm.ErrorView, "error.json", errors: ["A user with the specified `username` was not found"])
+          user ->
+            cond do
+              current_user_id == user_id ->
+                user = Users.unrestricted_changeset(user, params)
+                case Repo.update(user) do
+                  {:ok, result} ->
+                    conn
+                    |> put_status(:ok)
+                    |> render("private.json", user: result)
+                  {:error, changeset} ->
+                    conn
+                    |> put_status(:unprocessable_entity)
+                    |> render(Thegm.ErrorView, "error.json", errors: Enum.map(changeset.errors, fn {k, v} -> Atom.to_string(k) <> ": " <> elem(v, 0) end))
+                end
+              true ->
+                conn
+                |> put_status(:forbidden)
+                |> render(Thegm.ErrorView, "error.json", errors: ["You do not have privileges to edit this account"])
+            end
+        end
+      true ->
+        conn
+        |> put_status(:bad_request)
+        |> render(Thegm.ErrorView, "error.json", errors: ["Posted a non `user` data type"])
+    end
+  end
+
+  def show(conn, %{"username" => username}) do
```

Comment:
Is this right? Shouldn't we still be hitting the API with the user ID instead of the username?

Reply:
If so then I have misunderstood, this already went through in PR #41 I merely changed variable names to reflect the truth

Reply:
I reverted it over the weekend with 5b2d3c6f66acca6997c4b2572abfd56e9a9653df

Reply:
The API should always assume UUIDs. The website is able to calculate the UUID from the username or slug if it does not have the UUID

Reply:
Yup, based on the call @xlexi and I just had, the API should expect to receive a valid v5 UUID rather than a username.

Reply:
Changes have been made, requesting new review